osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -37,7 +37,7 @@ public class SelectionHandler : CompositeDrawable, IKeyBindingHandler<PlatformAc
 
         public int SelectedCount => selectedBlueprints.Count;
 
-        public IEnumerable<HitObject> SelectedHitObjects => selectedBlueprints.Select(b => b.HitObject);
+        public IEnumerable<HitObject> SelectedHitObjects => EditorBeatmap.SelectedHitObjects;
```

Comment:
I'm not even honestly sure exposing as `public` even makes sense anymore. Only usages are `BlueprintContainer` (which already has a resolved `EditorBeatmap`), and subclasses (which can either access the protected `EditorBeatmap`, or we could make it a convenience protected property if that's better for code quality)..

Reply:
I'd agree with removing and referencing the beatmap directly, if that's what you're suggesting.

Reply:
Can give that a try. I think this existed before `EditorBeatmap` was a thing.

Reply:
Seems to work fine.

Reply:
Are you sure about this? There's roughly 20-30 references to it.

Reply:
I didn't realise there were that many... Fine to leave it as a helper property in that case.

Reply:
Should I revert the changes then? I'm neutral on it.

Reply:
Personally I'm mostly worried about someone potentially accidentally getting selected hit objects through `SelectedBlueprints` (which is also public) instead of this and hitting this same bug again.